### DISCUSSION TITLE
Explicitly use `'ast` or `'arena` instead of `'a` for lifetime specifier for data structures related to arena.

### DISCUSSION THREAD

**Discussion by hyf0** [oxc-project/oxc#700] [General]
rustc use `'tcx` to explicitly show what data is related to arena.

See https://rustc-dev-guide.rust-lang.org/memory.html#the-tcx-and-how-it-uses-lifetimes for details.

Currently, oxc uses `'a` for lifetime specifier, which could be very misunderstood for '`a` is a commonly used name for lifetime specifier just like meaningless variable names `foo` or `bar`.

It's hard to distinguish whether a structure with lifetimes is due to hold data related to arena or just a reference.

**Comment by Boshen** [comment]
Close as not planned.
